src/shared/components/Datepicker/Datepicker.stories.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useRef, useState } from 'react'
+import { parse, isValid } from 'date-fns'
+import Datepicker, { DatepickerProps } from './Datepicker'
+import { Meta, Story } from '@storybook/react'
+
+export default {
+  title: 'Shared/Datepicker',
+  component: Datepicker,
+  args: {
+    helperText: 'Lorem ipsum dolor sit amet...',
+  },
+} as Meta
+
+const Template: Story<DatepickerProps> = (args) => {
+  const ref = useRef<HTMLInputElement | null>(null)
+  const [date, setDate] = useState<Date>()
+  const [validationError, setValidationError] = useState(false)
+
+  const handleInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void = (e) => {
+    const date = parse(e.target.value, 'dd/MM/yyyy', new Date())
```

Comment:
I think this specific line should live inside the component actually and the `date` should be passed as an argument to the `onChange`. Right now, we're coupling how the datepicker works to all the consumer. Let's say we want to support a different date format later on and we'd need to update all the consumers. If we parse, inside the component, then we only need to update the datepicker itself